1.Exploratory_Data_Analysis/scripts/2b.correlation_agg_morphology_features.py

```diff
@@ -0,0 +1,200 @@
+    feature_df["Metadata_inducer2"].notnull(),
+]
+results = [
+    (feature_df["Metadata_inducer1"]).astype(str),
+    (feature_df["Metadata_inducer1"] + "_" + feature_df["Metadata_inducer2"]).astype(
+        str
+    ),
+]
+feature_df["Metadata_Treatment"] = np.select(condlist=conditions, choicelist=results)
+
+# dose column merge
+conditions = [
+    (feature_df["Metadata_inducer2"].isnull()),
+    feature_df["Metadata_inducer2"].notnull(),
+]
+
+results = [
+    (feature_df["Metadata_inducer1_concentration"].astype(str)).astype(str),
+    (
+        feature_df["Metadata_inducer1_concentration"].astype(str)
+        + "_"
+        + feature_df["Metadata_inducer2_concentration"].astype(str)
+    ).astype(str),
+]
+feature_df["Metadata_Dose"] = np.select(condlist=conditions, choicelist=results)
+feature_df["Metadata_inducer1_concentration"] = pd.to_numeric(
+    feature_df["Metadata_inducer1_concentration"]
+)
+# one beta of inudcer1, inducer1 concentration, inhibitor, and inhibitor concentration all as 1 beta term
+feature_df["oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = (
+    feature_df["Metadata_Treatment"]
+    + "_"
+    + feature_df["Metadata_Dose"].astype(str)
+    + "_"
+    + feature_df["Metadata_inhibitor"].astype(str)
+    + "_"
+    + feature_df["Metadata_inhibitor_concentration"].astype(str)
+).astype(str)
+
+
+# ## Correlation of Aggregated Features per Well
+
+# In[5]:
+
+
+# drop metadata columns if the column contains 'Metadata'
+cell_df = feature_df
+feature_df = cell_df.loc[:, ~cell_df.columns.str.contains("Metadata")]
+metadata_df = cell_df.loc[:, cell_df.columns.str.contains("Metadata")]
+
+
+# In[6]:
+
+
+feature_df.loc[:, "Metadata_Well"] = metadata_df["Metadata_Well"]
+feature_df.index = feature_df["Metadata_Well"]
+feature_df = feature_df.drop(columns=["Metadata_Well"])
+
+
+# In[7]:
+
+
+well_corr_df = feature_df.T.corr()
+save_path = pathlib.Path(f"./results/correlation/{cell_type}/aggregated_morphology")
+save_path.mkdir(parents=True, exist_ok=True)
+well_corr_df.to_csv(f"{save_path}/wells_corr.csv")
+
+
+# # Correlation of Aggregated Features per Treatment
+
+# In[8]:
+
+
+feature_df = cell_df.loc[:, ~cell_df.columns.str.contains("Metadata")]
+metadata_df = cell_df.loc[:, cell_df.columns.str.contains("Metadata")]
+feature_df.loc[:, "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"] = metadata_df[
+    "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
+]
+# groupby oneb_Metadata_Treatment_Dose_Inhibitor_Dose column
+feature_df = feature_df.groupby("oneb_Metadata_Treatment_Dose_Inhibitor_Dose").median()
+
+
+# In[9]:
+
+
+well_corr_df = feature_df.T.corr()
+save_path = pathlib.Path(f"./results/correlation/{cell_type}/aggregated_morphology")
+save_path.mkdir(parents=True, exist_ok=True)
+well_corr_df.to_csv(f"{save_path}/treatments_corr.csv")
+
+
+# # Selected Treatment correlation
+
+# In[10]:
+
+
+treatments_agg = feature_df
+treatments_agg = treatments_agg.reset_index()
+
+
+# In[11]:
+
+
+list_of_treatments = [
```

Comment:
it would be difficult to maintain this list in all three locations.

Also, consider adding a comment about why to focus on these treatments specifically.